src/Certify.Core/Management/Challenges/DNS/DnsProviderPoshACME.cs

```diff
@@ -51,6 +53,37 @@ Implemented providers (Posh-ACME: https://github.com/rmbolger/Posh-ACME)
             [Zonomi](https://github.com/rmbolger/Posh-ACME/blob/master/Posh-ACME/DnsPlugins/Zonomi-Readme.md)
         */
 
+        public class PoshACMEDnsProviderProvider : IDnsProviderProviderPlugin
+        {
+            public IDnsProvider GetProvider(Type pluginType, string id)
+            {
+                foreach (var provider in ExtendedProviders)
+                {
+                    if (provider.Id == id)
+                    {
+                        var scriptPath = Path.Combine(Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location), @"Scripts\DNS\PoshACME");
+                        // TODO : move this out, shared config should be injected
```

Comment:
Yikes, this does suggest we need to pass this config in rather than refer to it here. You'll have seen that this one works by providing a bunch of Challenge Provider Definitions that all use the same type but have different definitions for parameters etc.

Reply:
It looks it's just trying to get the global PowerShell Execution Policy here. I just cut and pasted what was in `ChallengeProvider.cs`, including that comment.